pkg/driver/constants.go

```diff
@@ -87,4 +87,6 @@ const (
 	// Maximum default number of volumes that controller can publish to the node.
 	defaultMaxVolPerNode = 100
 	maxVolumesPerNodeKey = "MAX_VOLUMES_PER_NODE"
+	// GetNodeInfo polling frequency
+	PollingFrequencyInMin = 30
```

Comment:
Where are we getting the 30 minute interval from? How expensive is this? Can we poll every minute?

What I'm getting is that if there's a network change, that change will immediately be tested by an end-user and they might start digging themselves a hole because no one is going to sit on their hands for 30 minutes nor read the documentation to understand this behaviour. If we poll every minute, a few retries down the road and you're up and running.

Reply:
Polling every minute may put extra load on the cluster node if there is no frequent network change. 30 min is default polling delay as you see in the code.  I agree that any change to network configuration will be picked only after 30 min which may not be acceptable to the user.
We could add a configurable env variable for this. I will let @rgcostea  comment on this. thanks.

Reply:
So I came across fsnotify last night: https://godoc.org/github.com/fsnotify/fsnotify

Could we use this to set a watcher on /etc/sysconfig/network and/or /etc/sysconfig/network-scripts and/or any other files we think might change as users update their networks?  This way, it is notification driven rather polled.

Reply:
It is a better approach, but i think more challenging. We would need watch list of directories based on distro(/etc/network, /etc/sysconfig/network, /etc/iscsi/) for changes. The challenge here is some tool on host should not be generating spurious updates very often to any files under these without actual network change.

Reply:
Even if spurious updates come along, that doesn't mean we have to update the node info on each one.  We could be selective.  For example, we could read the files of interest and compare against our current node to determine if we need to perform an update.  Furthermore, we could coalesce multiple updates into one operation on our end.

Reply:
sure, sounds good.

Reply:
The fsnotify looks like a promising package that is os event-driven and useful for monitoring any file changes on host os that is different then the original requirement (captured in the Bug) to poll for networking changes and update the HPENodInfo.  We should evaluate the following points before we take this route : 
1. HPENodeInfo doesn't not only contain networking , it also contains  initiators name, domain name and chap info, is this requirement to monitor all of them or only networking ? 
2.  If there is any change in default configuration file path on host , this will not work.  I don't think there would be lot of such users.

Reply:
I'm working on this change.

Reply:
Yes, we should monitor for every property that can change including IQN and WWPN details.

Item 2 is a corner case as you say.  So in those cases, we just fall back to our current approach of restarting the node driver.

Reply:
Thanks @rgcostea . Done.

Reply:
@vidhutsingh please add directories /etc/network, /etc/sysconfig-networks, /etc/network-scripts and file /etc/iscsi/InitiatorName and remove /etc/iscsi from monitoring. /etc/iscsi will have iscsi DB on Ubuntu which gets updated on every volume attach/detach, but we just need to capture IQN change. We also discussed that we need rate limiting on this(i.e process only one change request every 5 minutes etc). For FC, i don't think we need monitoring as new adapter(WWN) involves host reboot anyway.

Reply:
Thanks shiva. sure I will make watcher list changes. 
Regarding, If we rate limit to 5 min than how different is this implementation than earlier polling implementation?

Reply:
It's different from polling because we would still only update when there is a change to these directories.  If no changes are made from 10pm to 8am the next morning, we wouldn't do anything.  The same cannot be said for polling.

Reply:
Okay, thanks for the clarification. so basically we should go to sleep mode for 5 min after invoking t.watchRun() . 
So after every event notification we will run the nodeGetInfo() and go to sleep for 5 min and if there is any event in between that will be ignored. That will rate limit.

Reply:
The corner case still exist, what if there is a genuine network update within 5 min window? that will be ignored. shall we stick to 1 min instead of 5 min? Thought?

Reply:
@shiva, we need /etc/iscsi/iscsid.conf file for CHAP update.

Reply:
Where did 5 minutes come from?  :)

I think 1 minute would be better.

Reply:
It's been suggested above. thanks for the confirmation.